Modix.Services/Moderation/MessageContentCheckBehaviour.cs

```diff
@@ -0,0 +1,134 @@
+        private readonly IModerationService _moderationService;
+        private readonly IMessageContentPatternService _messageContentPatternService;
+        private readonly IDiscordSocketClient _discordSocketClient;
+
+        public MessageContentCheckBehaviour(
+            IDesignatedChannelService designatedChannelService,
+            IDiscordSocketClient discordSocketClient,
+            IAuthorizationService authorizationService,
+            IModerationService moderationService, IMessageContentPatternService messageContentPatternService)
+        {
+            _designatedChannelService = designatedChannelService;
+            _discordSocketClient = discordSocketClient;
+            _authorizationService = authorizationService;
+            _moderationService = moderationService;
+            _messageContentPatternService = messageContentPatternService;
+        }
+
+        public Task HandleNotificationAsync(MessageReceivedNotification notification,
+            CancellationToken cancellationToken = default)
+            => TryCheckMessageAsync(notification.Message);
+
+        public Task HandleNotificationAsync(MessageUpdatedNotification notification,
+            CancellationToken cancellationToken = default)
+            => TryCheckMessageAsync(notification.NewMessage);
+
+        private async Task TryCheckMessageAsync(IMessage message)
+        {
+            if (!(message.Author is IGuildUser author)
+                || (author.Guild is null)
+                || !(message.Channel is IGuildChannel channel)
+                || (channel.Guild is null))
+            {
+                Log.Debug(
+                    "Message {MessageId} was not in an IGuildChannel & IMessageChannel, or Author {Author} was not an IGuildUser",
+                    message.Id, message.Author.Id);
+                return;
+            }
+
+            if (author.Id == _discordSocketClient.CurrentUser.Id)
+                return;
+
+            var isContentBlocked = await IsContentBlocked(channel, message);
+
+            if (!isContentBlocked)
+            {
+                return;
+            }
+
+            if (await _designatedChannelService.ChannelHasDesignationAsync(channel.Guild,
+                channel, DesignatedChannelType.Unmoderated, default))
+            {
+                return;
+            }
+
+            if (await _authorizationService.HasClaimsAsync(author, AuthorizationClaim.BypassMessageContentPatternCheck))
+            {
+                Log.Debug("Message {MessageId} was skipped because the author {Author} has the {Claim} claim",
+                    message.Id, message.Author.Id, AuthorizationClaim.BypassMessageContentPatternCheck);
+                return;
+            }
+
+            Log.Debug("Message {MessageId} is going to be deleted", message.Id);
+
+            await _moderationService.DeleteMessageAsync(message, "Unauthorized Message Content Link",
```

Comment:
Minor, but these aren't necessarily links anymore.

```suggestion
            await _moderationService.DeleteMessageAsync(message, "Unauthorized Message Content",
```